joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala

```diff
@@ -0,0 +1,211 @@
+package io.joern.go2cpg.passes.ast
+
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+import io.shiftleft.semanticcpg.language.*
+
+import scala.collection.immutable.List
+
+class StructureCallTests extends GoCodeToCpgSuite(withOssDataflow = true) {
+
+  "when structure declared with multiple argument" should {
+    val cpg = code(
+      """
+        package main
+        |func main() {
+        |    rect := Rectangle{width: 10, height: "somestr"}
+        |}
+        |""".stripMargin,
+      "test.go"
+    )
+    val List(rectangleCallNode) = cpg.call.nameExact("Rectangle").l
+    "test basic cpg properties for call node of structure declaration" in {
+      rectangleCallNode.name shouldBe "Rectangle"
+      rectangleCallNode.methodFullName shouldBe "main.Rectangle"
+      rectangleCallNode.typeFullName shouldBe "main.Rectangle.<init>"
+    }
+
+    "test argument of structure declaration node(call node)" in {
+      val List(widthNode, heightNode) = rectangleCallNode.argument.isLiteral.l
+      widthNode.typeFullName shouldBe "int"
+      heightNode.typeFullName shouldBe "string"
+    }
+
+    "test argument index" in {
+      val List(widthNode, heightNode) = rectangleCallNode.argument.isLiteral.l
+      widthNode.argumentIndex shouldBe 1
+      heightNode.argumentIndex shouldBe 2
+    }
+
+  }
+
+  "when structure declared with no argument" should {
+    val cpg = code(
+      """
+        package main
+        |func main() {
+        |    rect := Rectangle{}
+        |}
+        |""".stripMargin,
+      "test.go"
+    )
+
+    "check call node present" in {
+      cpg.call.nameExact("Rectangle").size shouldBe 1
+    }
+  }
+
+  "when is structure imported from other package structure" should {
+    val cpg = code(
+      """
+        package main
+        |import (
+        |    "path/to/somepackage"
+        |)
+        |func main() {
+        |    rect := somepackage.Rectangle{l: 10, w: 20}
+        |}
+        |""".stripMargin,
+      "test.go"
+    )
+    val List(rectangleCallNode) = cpg.call.nameExact("Rectangle").l
+    "test basic cpg properties for call node of structure declaration" in {
+      rectangleCallNode.name shouldBe "Rectangle"
+      rectangleCallNode.methodFullName shouldBe "path/to/somepackage.Rectangle"
+      rectangleCallNode.typeFullName shouldBe "path/to/somepackage.Rectangle.<init>"
```

Comment:
The constructor is not a type, and rather the `methodFullName` should be `path/to/somepackage.Rectangle.<init>`

Reply:
Thanks for pointing it out.
Corrected here: https://github.com/joernio/joern/pull/3596/commits/4d71b6e52c2b74cc54001de2a5eb74427307d804